raft-tech/TANF-app
# 2408.3 - refactor tanf t1 parsing

## Summary of Changes
Pull request closes #2408  

* Refactor implementation of `tanf.T1` to utilize the new `RowSchema` utility class.
* Implement the new schema in the `parse_datafile` function
* Update automated tests

This PR is an example of how to implement future parsing schemas within the new structure.

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && npm run start
cd tdrs-backend && docker-compose up
```

1. Open http://localhost:3000/ and sign in.
2. Grant your user an "Approved" account status and assign them to a "Data Analyst" role.
3. Navigate to the Data Files page and upload `small_correct_file` from `tdpservice/parsing/test/data` to "Section 1" for any quarter and any year.
4. Navigate to the django admin and view the newly created "T1" document. Make sure field values match expected values.

This PR does not introduce validation of the T1 document, which will be implemented with #2369.

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
